### ISSUE TITLE
ecma scm doesn't map private properties to highlight scopes

### ISSUE THREAD

**Issue by mesmere** [helix-editor/helix#11443]
### Summary

The official JavaScript tree-sitter scopes private properties of ES6 classes as:

||Scopes|
|-|-|
|Private fields:|`... field_definition, private_property_identifier`|
|Private methods:|`... method_definition, private_property_identifier`|

However, in Helix these aren't syntax highlighted and `:tree-sitter-highlight-name` reveals that they get no highlight scope at all.

- [ ] Private fields should get the `variable.other.member` highlight (like public fields do).
- [ ] Private methods should get the `function` highlight (like public methods do).

### Reproduction Steps

```js
class MyClass {
  field;
  method() {
    return;
  }

  #privateField;
  #privateMethod() {
    return;
  }
}
```

![swappy-20240807_135830](https://github.com/user-attachments/assets/a17581e8-f01b-42c4-b2d9-469b613ee5ba)

### Platform

arch

### Terminal Emulator

foot 1.17.2

### Installation Method

manjaro pacman repo

### Helix Version

helix 24.3

**Comment by mesmere** [comment]
It looks like someone else took a crack at this already with https://github.com/helix-editor/helix/pull/10554 but something's clearly broken.

**Comment by David-Else** [comment]
>  Private fields should get the variable.other.member highlight (like public fields do).
Private methods should get the function highlight (like public methods do).

![image](https://github.com/user-attachments/assets/4c10e724-4866-490a-8067-355ce9b4460f)
![image](https://github.com/user-attachments/assets/77adf88c-cda9-40d9-9f13-6a2136b45338)

This seems to already be the case, at least in `helix 24.7 (cfe80acb)`. I am using the `dark_plus` theme.

**Comment by the-mikedavis** [comment]
Yeah #10554 fixed this but you're on 24.03. If you update to 24.07 you should see that highlighting

**Comment by mesmere** [comment]
Oh, I'm sorry. I feel silly :X
